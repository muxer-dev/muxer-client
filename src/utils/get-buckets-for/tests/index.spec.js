import eventsForEveryBuckets from './test-data/events-for-every-buckets';
import eventsForSomeBuckets from './test-data/events-for-some-buckets';
import getBucketsFor from '../';

jest.mock('moment', () => (input) => {
  return require.requireActual('moment')(input || '2017-06-15T08:00:00');
});

describe('get-buckets-for', function () {
  it('should create empty buckets', function () {
    expect(getBucketsFor([])).toMatchSnapshot();
  });

  it('should add events to every bucket', function () {
    expect(getBucketsFor(eventsForEveryBuckets)).toMatchSnapshot();
  });

  it('should add events to some buckets', function () {
    expect(getBucketsFor(eventsForSomeBuckets)).toMatchSnapshot();
  });
});
