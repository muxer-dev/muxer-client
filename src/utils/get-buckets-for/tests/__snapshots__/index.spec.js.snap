// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get-buckets-for should add events to every bucket 1`] = `
Array [
  Object {
    "className": "events-today",
    "events": Array [
      Object {
        "id": "1",
        "name": "Meetup coming soon",
        "start": "2017-06-15T22:45:00",
      },
    ],
    "id": "today",
    "message": "Today",
  },
  Object {
    "className": "events-in-next-seven-days",
    "events": Array [
      Object {
        "id": "2",
        "name": "Meetup this week",
        "start": "2017-06-16T09:45:00",
      },
    ],
    "id": "in-next-seven-days",
    "message": "Next 7 days",
  },
  Object {
    "className": "events-in-next-thirty-days",
    "events": Array [
      Object {
        "id": "3",
        "name": "Meetup next month",
        "start": "2017-07-13T11:30:00",
      },
    ],
    "id": "in-next-thirty-days",
    "message": "Next 30 days",
  },
  Object {
    "className": "events-in-next-three-months",
    "events": Array [
      Object {
        "id": "4",
        "name": "Meetup in a few months",
        "start": "2017-09-20T19:00:00",
      },
    ],
    "id": "in-next-three-months",
    "message": "Next 3 months",
  },
  Object {
    "className": "events-in-the-future",
    "events": Array [
      Object {
        "id": "5",
        "name": "Meetup this year",
        "start": "2017-11-01T10:47:00",
      },
      Object {
        "id": "6",
        "name": "Meetup next year",
        "start": "2018-02-07T14:03:00",
      },
    ],
    "id": "in-the-future",
    "message": "And beyond...",
  },
]
`;

exports[`get-buckets-for should add events to some buckets 1`] = `
Array [
  Object {
    "className": "events-today",
    "events": Array [
      Object {
        "id": "1",
        "name": "Meetup coming soon",
        "start": "2017-06-15T22:45:00",
      },
    ],
    "id": "today",
    "message": "Today",
  },
  Object {
    "className": "events-in-next-seven-days",
    "events": Array [
      Object {
        "id": "2",
        "name": "Meetup this week",
        "start": "2017-06-16T09:45:00",
      },
    ],
    "id": "in-next-seven-days",
    "message": "Next 7 days",
  },
  Object {
    "className": "events-in-next-thirty-days",
    "events": Array [],
    "id": "in-next-thirty-days",
    "message": "Next 30 days",
  },
  Object {
    "className": "events-in-next-three-months",
    "events": Array [
      Object {
        "id": "3",
        "name": "Meetup this week too",
        "start": "2017-07-16T11:30:00",
      },
      Object {
        "id": "4",
        "name": "Meetup in a few months",
        "start": "2017-09-20T19:00:00",
      },
      Object {
        "id": "5",
        "name": "Meetup later this year",
        "start": "2017-09-22T10:47:00",
      },
    ],
    "id": "in-next-three-months",
    "message": "Next 3 months",
  },
  Object {
    "className": "events-in-the-future",
    "events": Array [
      Object {
        "id": "6",
        "name": "Meetup next year",
        "start": "2018-02-07T14:03:00",
      },
    ],
    "id": "in-the-future",
    "message": "And beyond...",
  },
]
`;

exports[`get-buckets-for should create empty buckets 1`] = `
Array [
  Object {
    "className": "events-today",
    "events": Array [],
    "id": "today",
    "message": "Today",
  },
  Object {
    "className": "events-in-next-seven-days",
    "events": Array [],
    "id": "in-next-seven-days",
    "message": "Next 7 days",
  },
  Object {
    "className": "events-in-next-thirty-days",
    "events": Array [],
    "id": "in-next-thirty-days",
    "message": "Next 30 days",
  },
  Object {
    "className": "events-in-next-three-months",
    "events": Array [],
    "id": "in-next-three-months",
    "message": "Next 3 months",
  },
  Object {
    "className": "events-in-the-future",
    "events": Array [],
    "id": "in-the-future",
    "message": "And beyond...",
  },
]
`;
